### PR TITLE
Expose preferences via StytchClient

### PR DESCRIPTION
**Pull request by TomMichalkevic**
Linear Ticket: [MOB-99](https://linear.app/stytch/issue/MOB-99)

## Changes:

1. Exposing map of preferences/cookies in StytchClient

### CODE REVIEW

**Comment by jhaven-stytch** [comment]
@TomMichalkevic is there a linear ticket/some context around this change? Trying to get caught up on work-in-flight, and not sure what this relates to. Thanks!

**Comment by TomMichalkevic** [comment]
@jhaven-stytch I've added a Linear ticket to give some context. Apologies for not adding it earlier :)

**Review by jhaven-stytch — APPROVED** [review]
Ok, lgtm then!

Apologies that my merge caused conflicts 😅

**Comment by TomMichalkevic** [comment]
No problem ;) Squashed the commits and resolved the conflicts.

File: sdk/src/main/java/com/stytch/sdk/StorageHelper.kt [hunk 1]
```diff
@@ -52,6 +52,23 @@ internal object StorageHelper {
...
 
+    /**
+     * Load and return a Map of all cookies in memory
+     * @param Map<String, String> of cookie and preference names
+     * @return Map<String?, String?> of names and values of cookies saved in memory or null if there are none
+     * @throws Exception if failed to return data
+     */
+    internal fun getAllCookies(cookiePrefMap: Map<String, String>): Map<String?, String?>? {
+        return try {
+            cookiePrefMap.mapValues {
```
[the remark this answers is not in the document]
Comment (dan-stytch) [reply sdk/src/main/java/com/stytch/sdk/StorageHelper.kt:63]: Also, if we're making this really seamless for our sdk consumers, perhaps we want to include these values as well?

https://github.com/stytchauth/stytch-swift/blob/main/Sources/StytchCore/SessionStorage.swift#L117-L127

Maybe that would make this function return a `Map<Map<String, String>>?` where you'd return the cookie name as the top level key, with the cookie keys/values as the map values within that? Does that make sense? Also worth considering whether we provide consumers with the ability to "subscribe" to cookie updates, so they get notified when these cookie values change.

**Review by dan-stytch — COMMENTED** [review]
I think there is more we can do (added comments/questions), but lmk if you think we should keep it simple for now

File: sdk/src/main/java/com/stytch/sdk/StorageHelper.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhaven-stytch) [reply sdk/src/main/java/com/stytch/sdk/StorageHelper.kt:63]: @dan-stytch we should sync about this and what the customer usecase is, as I'm still not 100% sure on that. Once I fully understand it I can take another pass at this 👍 Making this an observable/flow is probably possible, but again, wanna fully grok it first :)

**Comment by jhaven-stytch** [comment]
Closing in preference of SDK-640

**Pull request closed without merging**